src/components/definition-list/dl.component.js

```diff
@@ -4,27 +4,48 @@ import PropTypes from 'prop-types';
 import { StyledDl, StyledDtDiv, StyledDdDiv } from './definition-list.style';
 import Dt from './dt.component';
 import Dd from './dd.component';
+import createGuid from '../../utils/helpers/guid';
 
 const Dl = ({
   children,
   w = 50,
   dtTextAlign = 'right',
   ddTextAlign = 'left'
 }) => {
-  const dtLabels = React.Children.toArray(children).filter(child => child.type === Dt);
-  const ddContent = React.Children.toArray(children).filter(child => child.type === Dd);
+  const dlComponent = [];
+  const listChildren = React.Children.toArray(children);
+  let dtLabel;
+  let ddContent = [];
 
-  return (
-    <StyledDl data-component='dl'>
+  listChildren.forEach((child, index) => {
+    if (child.type === Dt) {
+      dtLabel = child;
+    }
+    if (child.type === Dd) {
+      ddContent.push(child);
+    }
 
-      <StyledDtDiv w={ w } dtTextAlign={ dtTextAlign }>
-        {dtLabels}
-      </StyledDtDiv>
+    if (dtLabel && (index === (listChildren.length - 1) || listChildren[index + 1].type === Dt)) {
+      const key = child.props.key || `dl_row_${createGuid()}}`;
+      dlComponent.push(
+        <React.Fragment key={ key }>
```

Comment:
`createGuid` has the same problem as using `index` in this case, as the order doesn't change, it's fine to use `index`.

https://reactjs.org/docs/lists-and-keys.html#keys

```suggestion
        <React.Fragment key={ child.props.key || index }>
```

Reply:
ok thanks